staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto

```diff
@@ -1502,3 +1507,70 @@ message ReopenContainerLogRequest {
 
 message ReopenContainerLogResponse{
 }
+
+// Common options used for checkpointing and restoring.
+message CheckpointRestoreOptions {
+    // Keep temporary files. Like log files. Helpful for debugging.
```

Comment:
What would it look like to add additional options or override existing ones? One prerequisite for checkpoint/restore to work is for CRIU to be available and so options in the /etc/criu/default.conf would be honored? But one drawback with this would be that it would apply to all containers and options might differ depending on the specific container. Should this be made part of container spec somehow? Any thoughts on overall CRIU configurations for specific container?

Reply:
This questions feels out of scope of this PR. For the current use case and the goal of this PR (minimal checkpoint support and CRI API changes to start implementation on the container engine level) I do not think it is necessary to already have a clear definition how to handle additional parameters.

There is, however, already a runc annotation which supports specifying a CRIU configuration file in the container's `config.json` file. If we enable this in the container engine level, to specify per container CRIU options we can extend the CRI API, if necessary, to pass container specific CRIU options from the kubelet to the container engine via the CRI API.

Reply:
I like the idea of extending CRI API to specify per container CRIU options from kubelet to the container engine and I can understand this being out of this PR. Is there a way to track these thoughts/ideas somewhere or maybe this could be a 1 liner comment so that we don't lose track of them?